ServiceStack/src/ServiceStack/Validation/ValidationFilters.cs

```diff
@@ -95,8 +95,16 @@ await HostContext.RaiseServiceException(req, requestDto, validationResult.ToExce
                 {
                     errorResponse = validationFeature.ErrorResponseFilter(req, validationResult, errorResponse);
                 }
-    
-                await res.WriteToResponse(req, errorResponse);
+
+                // If the response is null, it means the request is sent from the InProcessServiceGateway
+                if (res == null)
+                {
+                    req.Items[Keywords.ServiceGatewayResponseStatus] = errorResponse.GetResponseStatus();
```

Comment:
We stuff the ResponseStatus into the Items dictionary because later on we will pull it out and set it on the response dto